### DISCUSSION TITLE
Language detection on streaming

### DISCUSSION THREAD

**Discussion by gdelco** [deepgram/community#466] [Feature Requests]
Anyone knows when this feature (language detection on streaming) will be added?
Thanks in advance.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by nikolawhallon** [reply]
It is something we have been thinking about, but we have no timeline to share at the moment. I'm personally pretty interested in this use-case and am curious how you would like to use it - word-by-word? sentence-by-sentence?

**Reply by gdelco** [reply]
The idea is to use it with twillio phone calls, so we can know which is the language of the person since the start. For this I do not think word-by-word is that useful, as I am going to receive all the sentence.

**Reply by nikolawhallon** [reply]
I've experimented with something similar - that is, trying to determine at the start of a call/stream the language and then apply that to the whole call - one issue that comes up with phone calls is often automated systems will begin the call in one language (e.g. English) and then once the caller starts speaking, the call switches to another language (e.g. Spanish) - in those cases, determining the language at the beginning of the call ruins the results for the rest of the call. Just curious if that is a concern.

But it sounds like the use-case is streaming and single-language detection? Definitely something we are/will look into, but sorry to say no timeline at the moment.

**Reply by bendixBoxer** [reply]
I am interested in this too. We have our own fix for this now, but would make more sense it coming from you. if you want to understand what we are interested in, let me know @nikolawhallon

**Reply by TasseDeCafe** [reply]
I'd also be interested in this feature. My use case here would be for a language app where the users can switch between different languages (typically the language they're learning and their native language). Transcription in multiple languages is something that doesn't work well with Whisper.

**Reply by A3Ackerman** [reply]
Hey, just wanted to chime in that my business would also have a use case for language detection w/ streaming for voice calls. The call can start with a message like "Hello, Bonjour" or "Bienvenidos, Welcome" which are common phrases used in bilingual cities to indicate that the person providing a service is happy to continue in either language. We'd like to be able to use this pattern for our AI voice agents that are using Deepgram transcription.
